addon/-private/transforms/boolean.js

```diff
@@ -23,9 +27,17 @@ import Transform from "ember-data/transform";
   @namespace DS
  */
 export default Transform.extend({
-  deserialize(serialized) {
+  deserialize(serialized, options) {
     var type = typeof serialized;
 
+    if (isEnabled('ds-transform-pass-options')) {
+      if (isEnabled('ds-boolean-transform-allow-null')) {
+        if (isNone(serialized) && options.allowNull === true) {
```

Comment:
Theoretically `options` should never be `undefined` because of [this](https://github.com/emberjs/data/blob/c78d7f0ea9b12c077dd6573b6d9e01e9ee1d2d70/addon/serializers/json.js#L201) and ultimately because of [this](https://github.com/emberjs/data/blob/48a0094f719b39c3dfbe48884cf0b7cbb930e5b2/addon/attr.js#L94).

It would only be `undefined` if the transform is used outside of regular `ember-data` usage...